src/svc_auth_gss.c

```diff
@@ -678,6 +678,8 @@ svcauth_gss_destroy(SVCAUTH *auth)
 		gss_release_buffer(&min_stat, &gd->pac.ms_pac);
 
 	gss_release_buffer(&min_stat, &gd->checksum);
+
+	mutex_unlock(&gd->lock);
```

Comment:
Well, the obvious thing, ithe lock protects the gd structure from being changed somewhere else while we're doing it.
gss_release_name() and gss_release_buffer() both change attributes of the gd structure.